app/config/api.py

```diff
@@ -19,7 +22,117 @@
 
 logger = getLogger(__name__)
 
-api = NinjaAPI()
+LogExtra = TypedDict(
+    'LogExtra',
+    {
+        'http': {
+            'request': {
+                'method': str, 'header': dict[str, str]
+            },
+            'response': {
+                'status_code': int, 'header': dict[str, str]
+            }
+        },
+        'url': {
+            'path': str, 'scheme': str
+        }
+    }
+)
+
+
+def generate_log_extra(request: HttpRequest, response: HttpResponse) -> LogExtra:
+    """Generate the extra dict for the logging calls
+
+    This will format the following extra fields to be sent to the logger:
+
+    request:
+        http:
+            request:
+                method: GET | POST | PUT | ...
+                header: LIST OF HEADERS
+            response:
+                header: LIST OF HEADERS
+                status_code: STATUS_CODE
+
+    url:
+        path: REQUEST_PATH
+        scheme: REQUEST_SCHEME
+
+    Args:
+        request (HttpRequest): Request object
+        response (HttpResponse): Response object
+
+    Returns:
+        dict: dict of extras
+    """
+    return {
+        'http': {
+            'request': {
+                'method': request.method or 'UNKNOWN',
+                'header': {
+                    k.lower(): v for k, v in request.headers.items()
```

Comment:
Does this log credentials?

Reply:
Very valid question. I don't see the api gateway headers in the logs, but I'm not sure if that's intended or not. I have to do some more checks.

Reply:
I think ATM there is nothing yet but I don't know what the plan is here.

Reply:
So we'll see `geoadmin-authenticated` which is as I understand it a `bool` and `geoadmin-username`. I think it might even be valuable to be able to see these?

Reply:
You're right.

But how about generally skipping (the content of) the [Authorization Header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization)? If we ever add Token or Basic Auth and forget to adapt the logging?